### ISSUE TITLE
Loading gif doesn't work in Safari

### ISSUE THREAD

**Issue by pcraig3** [HHS/simpler-grants-pdf-builder#115]
In most browsers, we see a little horse gif and the button change to say "Importing..." when we are importing (or re-importing) a NOFO.

![Image](https://github.com/user-attachments/assets/c0e01a72-c269-4e0e-9351-c44d1b73e9ea)

However, in Safari we don't see this.

This issue is not really about logging the issue so much as me explaining why it's a WONTFIX.

## Why this is not something that will be fixed 

So our JS loading animation triggers on this event: `form.addEventListener("submit")`.

One of the first things is does is to disable the button (this is so that people can't keep clicking the button while the form is submitting.

When investigating this issue, I changed the trigger to this: `button.addEventListener("click")`, and then the loading animation started to work.
So that _seemed_ good, but then the form wouldn't submit anymore. Presumably, this is because the button is being disabled.

But no worries, we can submit the form programmatically with `form.submit()`.  This works to submit the form, but then the loading gif does not show up again.

I put the two approaches here to ChatGPT and it gave me this as a plausible answer:

> When the form is submitted, the browser's default behavior is to send the request immediately, potentially cutting off the rendering pipeline for UI changes like adding the visible class to the GIF.

> This behavior can be less noticeable in other browsers if they render the DOM update before navigating away but might occur in Safari due to its rendering optimizations.

One if its suggestions was to do this:

```js
form.addEventListener("submit", function (event) {
  // Prevent immediate form submission
  event.preventDefault();

  // Show loading gif

  // Delay form submission slightly to allow for rendering
  setTimeout(() => {
    form.submit(); // Submit the form programmatically
  }, 500); // Adjust the delay as needed
});
```

In other words: 

- listen for the 'submit' event
- prevent it from submitting
- do all the loading gif stuff
- set a timeout to submit the form after .5 seconds

Now, this actually does work, but it is crazy that we would do this. Why would we intentionally delay the form submitting so that people can see the loading interaction??

So at this point, I decided that the loading interaction can be understood as an enhancement, and therefore, there is no more work to do here. If other browsers can see it, that's a bonus — but it is still 100% functional in safari. It just appears that once the form is submitted, the UI rendering on it stops.

**Comment by pcraig3** [comment]
@admoorgit, you just flagging you to acknowledge that this issue (a) exists, but (b) it is not worth the effort to fix it.
